src/tsd/RpcHandler.java

```diff
@@ -207,78 +148,158 @@ private void handleTelnetRpc(final Channel chan, final String[] command) {
   }
 
   /**
-   * Finds the right handler for an HTTP query and executes it.
-   * Also handles simple and pre-flight CORS requests if configured, rejecting
-   * requests that do not match a domain in the list.
+   * Creates a query instance capable of handling the given request.
+   * @param tsdb
+   * @param request
+   * @param chan
+   * @return a sublcass of {@link AbstractHttpQuery}
+   * @throws BadRequestException if the request is invalid in a way that
+   * can be detected early, here.
+   */
+  private AbstractHttpQuery createQueryInstance(final TSDB tsdb,
+        final HttpRequest request,
+        final Channel chan) 
+            throws BadRequestException {
+    final String uri = request.getUri();
+    if (Strings.isNullOrEmpty(uri)) {
+      throw new BadRequestException("Request URI is empty");
+    } else if (uri.charAt(0) != '/') {
+      throw new BadRequestException("Request URI doesn't start with a slash");
+    } else if (plugins_manager.isHttpRpcPluginPath(uri)) {
```

Comment:
Don't use regex to compare the paths of every HTTP request. A number of users are pushing data with the ``/api/put`` endpoint so we don't want to slow down their writes. It may be quicker to explode the URI and see if it matches ``PLUGIN_BASE_WEBPATH`` or just see if it contains the plugin base.